services/API-service/src/api/admin-area/admin-area.service.ts

```diff
@@ -132,32 +132,40 @@ export class AdminAreaService {
         'ST_AsGeoJSON(area.geom)::json As geom',
         'area."countryCodeISO3"',
       ])
-      .leftJoin(
-        AdminAreaDynamicDataEntity,
-        'dynamic',
-        'area.placeCode = dynamic.placeCode',
-      )
-      .addSelect([
-        'dynamic.value AS "population_affected"',
-        'dynamic."leadTime"',
-        'dynamic."date"',
-      ])
       .where('area."countryCodeISO3" = :countryCodeISO3', {
         countryCodeISO3: countryCodeISO3,
       })
-      .andWhere('dynamic."leadTime" = :leadTime', { leadTime: leadTime })
-      .andWhere('area."adminLevel" = :adminLevel', { adminLevel: adminLevel })
-      .andWhere('date = current_date');
+      .andWhere('area."adminLevel" = :adminLevel', { adminLevel: adminLevel });
 
-    const country = await this.countryRepository.findOne({
-      select: ['defaultAdminLevel'],
-      where: { countryCodeISO3: countryCodeISO3 },
-    });
-    // Only add triggered-area filter if this is the default admin level
-    if (placeCodes.length && adminLevel == country.defaultAdminLevel) {
-      adminAreasScript = adminAreasScript.andWhere(
-        'area."placeCode" IN (:...placeCodes)',
-        { placeCodes: placeCodes },
+    if (adminLevel == country.defaultAdminLevel) {
+      adminAreasScript = adminAreasScript
+        .leftJoin(
+          AdminAreaDynamicDataEntity,
+          'dynamic',
+          'area.placeCode = dynamic.placeCode',
+        )
+        .addSelect([
+          'dynamic.value AS "population_affected"',
+          'dynamic."leadTime"',
+          'dynamic."date"',
+        ])
+        .andWhere('dynamic."leadTime" = :leadTime', { leadTime: leadTime })
+        .andWhere('date = current_date');
+
+      const placeCodes = await this.getTriggeredPlaceCodes(
+        countryCodeISO3,
+        leadTime,
+      );
+
+      if (placeCodes.length) {
+        adminAreasScript = adminAreasScript.andWhere(
+          'area."placeCode" IN (:...placeCodes)',
+          { placeCodes: placeCodes },
+        );
+      }
+    } else {
```

Comment:
@gulfaraz It's especially this else-clause that I just meant with the hacky solution. It's a solution to errors that would otherwise appear in the front-end (about 'population_affected' missing.) It would be better to solve them there actually (I guess something with splitting the mapService.setAdminRegionStyle method between boundary layers and aggregates layers). But I did not have time to look at this properly.

Reply:
@jannisvisser I agree. Your commits are sufficient given our requirements. For a proper solution, this would need to be discussed in a new PBI/spike. Please feel free to create one if you want to prioritize it.